pkgs/top-level/all-packages.nix

```diff
@@ -3574,7 +3574,7 @@ let
     inherit fetchurl stdenv;
   });
 
-  gnu-smalltalk = callPackage ../development/compilers/gnu-smalltalk { };
+  gnu-smalltalk = callPackage ../development/compilers/gnu-smalltalk (config.gnu-smalltalk or {});
```

Comment:
Done, but may I ask why? I've seen it used elsewhere and it seems a bit cleaner to me than setting individually.